cmd/webhook.go

```diff
@@ -289,17 +301,58 @@ func createPatch(pod *corev1.Pod, sidecarConfigTemplate *Config, annotations map
 			sidecarConfig.Volumes[1].Name = csiEphemeralVolumeountName
 			sidecarConfig.Volumes[1].CSI.VolumeAttributes["fdPassingEmptyDirName"] = fdPassingvolumeMountName
 
+			// Add container and volume to the patch
 			patch = append(patch, addContainer(pod.Spec.Containers, sidecarConfig.Containers, "/spec/containers")...)
-
 			patch = append(patch, addVolume(pod.Spec.Volumes, sidecarConfig.Volumes, "/spec/volumes")...)
 			patch = append(patch, updateAnnotation(pod.Annotations, annotations)...)
 			patch = append(patch, updateWorkingVolumeMounts(pod.Spec.Containers, csiEphemeralVolumeountName, bucketMount, filerName, isFirstVol)...)
-			isFirstVol = false // update such that no longer the first value
+			isFirstVol = false // Update such that no longer the first value
 		}
 	}
+
 	return json.Marshal(patch)
 }
 
+// Function to clean and sanitize the name by removing illegal characters
+func cleanAndSanitizeName(name string) string {
+	// Define the allowed regex pattern: lowercase letters, numbers, and dashes
+	validNameRegex := regexp.MustCompile(`[^a-z0-9-]`)
+
+	// Replace any character that does not match the allowed pattern with an empty string
+	name = validNameRegex.ReplaceAllString(name, "")
+
+	// Replace double dashes with a single dash
+	name = strings.ReplaceAll(name, "--", "-")
+
+	// Remove trailing dashes
+	name = strings.TrimRight(name, "-")
+
+	return name
+}
+
+// Function to ensure name uniqueness by appending an integer if the name already exists
+func ensureUniqueName(baseName string, existingNames []string) string {
+	// Check if the name is already in the list of existing names
+	if slices.Contains(existingNames, baseName) {
+		// Append an integer to make the name unique
+		for i := 1; ; i++ {
+			newName := fmt.Sprintf("%s-%d", baseName, i)
```

Comment:
Isn't this function used to print formated strings? Does it actually return a string? Wouldnt it be more clear to do it like it used to be, like `filerBucketName = filerBucketName + "-" + strconv.Itoa(len(filerBucketList)+1)` , so in your case something like `newName := baseName + "-" + strconv.Itoa(i)`

Reply:
sprintf works.